sdk/src/hooks/useQuote.ts

```diff
@@ -1,132 +1,144 @@
 import { type UseQueryResult, useQuery } from '@tanstack/react-query'
 import { useMemo } from 'react'
-import { type Address, fromHex, toHex, zeroAddress } from 'viem'
-import type { Deposit, Expense } from '../types/order.js'
+import { type Address, fromHex, zeroAddress } from 'viem'
+import { toJSON } from './util.js'
+
+type Quoteable = { isNative: boolean; token?: Address; amount: bigint }
```

Comment:
IIRC what we discussed last week, I think this type could be refined to ensure the `token` address is always provided when `isNative` is `false`
```suggestion
type Quoteable =
  | { isNative: true; token?: never; amount: bigint }
  | { isNative: false; token: Address; amount: bigint }
```